test/common/verifier.go

```diff
@@ -140,5 +166,19 @@ func (v *Verifier) generateAndGetLogs(ctx context.Context, programName, severity
 }
 
 func (v *Verifier) constructRegex(programName, logMessage string) string {
-	return fmt.Sprintf(` %s %s %s .* %s\[\d+\]: .* %s`, v.projectName, v.shootName, v.shootUID, programName, logMessage)
+	expectedNodeHostName := v.nodeName
+	switch {
+	case v.providerType == "aws":
+		// On aws the the nodes ara named by adding the regional domain name after the instance, e.g.:
+		// `ip-xxx-xxx-xxx-xxx.ec2.<region>.internal`. However the rsyslog `hostname` property only returns the
+		// instance name - `ip-xxx-xxx-xxx-xxx`.
+		expectedNodeHostName = strings.SplitN(v.nodeName, ".", 2)[0]
+	case v.providerType == "alicloud":
+		// On alicloud the name of a node is made of lower case characters, e.g. `izgw846obiag360olq8sdaz`.
+		// However, the rsyslog `hostname` property can also contain upper case characters, e.g. `iZgw846obiag360olq8sdaZ`.
+		// This is why we use the (?i:...) - to turn on case-insensitive mode for the hostname matching.
+		expectedNodeHostName = "(?i:" + expectedNodeHostName + ")"
+	}
```

Comment:
A cosmetic suggestion that is up to you whether to incorporate or not:
```suggestion
	var expectedNodeHostName string

	switch {
	case v.providerType == "aws":
		// On aws the the nodes ara named by adding the regional domain name after the instance, e.g.:
		// `ip-xxx-xxx-xxx-xxx.ec2.<region>.internal`. However the rsyslog `hostname` property only returns the
		// instance name - `ip-xxx-xxx-xxx-xxx`.
		expectedNodeHostName = strings.SplitN(v.nodeName, ".", 2)[0]
	case v.providerType == "alicloud":
		// On alicloud the name of a node is made of lower case characters, e.g. `izgw846obiag360olq8sdaz`.
		// However, the rsyslog `hostname` property can also contain upper case characters, e.g. `iZgw846obiag360olq8sdaZ`.
		// This is why we use the (?i:...) - to turn on case-insensitive mode for the hostname matching.
		expectedNodeHostName = "(?i:" + expectedNodeHostName + ")"
	default:
		expectedNodeHostName = v.nodeName
	}
```